hydra-tui/src/Hydra/TUI.hs

```diff
@@ -47,23 +44,28 @@ import Lens.Micro.TH (makeLensesFor)
 import Paths_hydra_tui (version)
 import Shelley.Spec.Ledger.API (UTxO (..))
 import qualified Shelley.Spec.Ledger.API as Cardano
-import Test.QuickCheck.Gen (Gen (..), scale)
-import Test.QuickCheck.Random (mkQCGen)
 import qualified Prelude
 
 --
 -- Model
 --
 
 data State = State
-  { me :: Host
+  { me :: Maybe Party -- TODO(SN): we could make a nicer type if refactor like
+  -- below as we would know the party always when
+  -- 'Connected'
+  , nodeHost :: Host
   , peers :: [Host]
   , headState :: HeadState
   , dialogState :: DialogState
   , clientState :: ClientState
   , feedback :: Maybe UserFeedback
   }
 
+-- XXX(SN): This is what I had tried to avoid in the past.. refactor back into a
+-- "make impossible states unrepresentable". Allowing 'Disconnected' + other
+-- info we still have around in the 'State' brings us into the uncomfortable
+-- situation of needing to decide which of the values are not stale.
 data ClientState = Connected | Disconnected
```

Comment:
@KtorZ @abailly-iohk I would really like to this discuss this. Maybe tomorrow when most of the stress is over? :)

Reply:
You mean distinguishing the state through the constructors of `State`?